### DISCUSSION TITLE
Sharing data between livewire components

### DISCUSSION THREAD

**Discussion by longestdrive** [livewire/livewire#9021] [1. Help]
Hi
What's the best approach to sharing data between livewire components and correct place for a db call? I'm used to static pages with Laravel and getting confused with best practice with Livewire.

I have a page that has two elements I'm trying to convert into Livewire components. Component 1 is an individual leaderboard and component 2 is a teams leaderboard (each team made up from individual players that feature in the leaderboard). 

Both components need to pull the same data from the database but manipulate the data in different ways. The data will need refreshing based on interaction with these and other components - eg fetch updated scores.

I'm trying to avoid repeating the database call within each component and at the same time check my understanding of the `mount` and `render` methods to work out the best place for that database call.

What's the best approach to share the data? Options I was thinking about:

- Create a parent class that makes the db call and then shares that data with the two child classes (seems messy)
- Simply allow the 2 components to make the same db call
- Pass the data from one component to the next - but unsure how best to approach that (could I use an event with a data parameter?)

And in calling the data I created a method in the component that makes the call. This method instantiates a repository class - I call that method from within the `mount` method and then call it again at each response to an event triggered by another component when scores are updated etc? Can I place it in the render method and is it Ok to pass that to the view using a `with` method similar to a standard approach in Laravel? 

When the component responds to an event - I understand the mount method is not called again and only at initiation. Also that I would need to re-instantiate the repository class within the livewire component - is that correct?

Any advice to steer in the right direction appreciated :)

**Comment by lcallif** [comment]
Depending on the size, you could put it in cache.

```
        $this->refviews = cache()->remember('customerview',60*60*24*7, function(){
            return Customerview::orderby('theArea')
                ->orderby('shortDescription')
                ->get();
        });
```
